chat-plugins/chat-monitor.js

```diff
@@ -0,0 +1,217 @@
+'use strict';
+
+const FS = require('../lib/fs');
+
+const MONITOR_FILE = 'config/chat-plugins/chat-monitor.json';
+
+/** @type {{[k: string]: string[]}?} */
```

Comment:
...?

I'm actually kinda confused why `data` and `filterWords` need to be separate variables in the first place.

Reply:
Remnant from before I rewrote parts.

Reply:
Actually, it's necessary because data is possibly null, while we assume filterWords is never null.